app/models/manageiq/providers/redhat/infra_manager.rb

```diff
@@ -45,23 +48,34 @@ def supports_authentication?(authtype)
     supported_auth_types.include?(authtype.to_s)
   end
 
-  def self.raw_connect(server, port, username, password, service = "Service")
-    require 'ovirt'
+  def raw_connect(server, port, username, password, service = "Service")
+    # If there is API path stored in the endpoints table and use it:
+    path = default_endpoint.path
+    unless path.blank?
+      _log.info("Using stored API path '#{path}'.")
+    end
+
     Ovirt.logger = $rhevm_log
 
     params = {
       :server     => server,
       :port       => port.presence && port.to_i,
+      :path       => path,
       :username   => username,
       :password   => password,
       :verify_ssl => false
     }
 
-    read_timeout, open_timeout = ems_timeouts(:ems_redhat, service)
+    read_timeout, open_timeout = self.class.ems_timeouts(:ems_redhat, service)
     params[:timeout]      = read_timeout if read_timeout
     params[:open_timeout] = open_timeout if open_timeout
 
-    Ovirt.const_get(service).new(params)
+    result = Ovirt.const_get(service).new(params)
+
+    # Copy the API path to the endpoints table:
+    default_endpoint.path = result.api_path
```

Comment:
@Fryguy Are you suggesting that we update the endpoint every time we connect to the provider?